src/tribler-core/run_tribler_headless.py

```diff
@@ -68,12 +70,21 @@ def log_incoming_remote_search(self, sock_addr, keywords):
         signal.signal(signal.SIGINT, lambda sig, _: ensure_future(signal_handler(sig)))
         signal.signal(signal.SIGTERM, lambda sig, _: ensure_future(signal_handler(sig)))
 
-        config = TriblerConfig(options.statedir or Path(get_appstate_dir(), '.Tribler'))
+        root_state_dir = Path(options.statedir) if options.statedir else get_root_state_directory()
```

Comment:
I see you're copy-pasting this stuff from `run_tribler.py`. Maybe there is a way to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) this stuff by `import`ing it from `run_tribler.py`?

Reply:
may be I can use run_tribler.py directly but instead of using environment variables like

https://github.com/Tribler/tribler/blob/09fbd720a6c251ffa76bfb2d2560b60a6920994d/src/run_tribler.py#L107

I need to add argparser directly and switch to loading core only with this argument options or-ed with environment variables. This way I can also push apikey as well.

This way i ca DRY it however i need to change run_tribler.py, which is kinda fundemental I prevented this in the first phase.

What do you think? If you think its cool then this PR goes to depths of hell :)

Reply:
I was thinking about unifying `run_tribler` with `run_tribler_headless` too. Note though, that `run_tribler` is a top-level script. It is not packaged alongside `tribler_core`. On the other hand, `run_tribler_headless` belongs to `tribler_core` package.

 The correct way to solve this is to make `run_tribler` use `run_tribler_headless` as an external script. The biggest problem is in packaging. Let's merge this PR as it is.

Nonetheless, I dare you to try doing it the proper way (i.e. running `run_tribler_headless` from `run_tribler`) :wink:

Reply:
If you execuse me with the new test cases, i can do that, if there is already an existing test cases i can modify them though.

Projected files to modify:
Run_tribler.py
Run_tribler_headless.py
Triblerd.spec

Reply:
That's the problem - there are no real tests for these, except for deployment testers (which run only nightly and/or release builds).

If you figure out how to test these properly, that would be of great service to Tribler project!

Reply:
Hmm, i think best approach i do a fork my own from devel and work this case on my own branch.

i can keep up with the code itself but integrating tests is beyond my resource availability.